spec/lib/services/api/metric_rollups_service_spec.rb

```diff
@@ -0,0 +1,19 @@
+describe Api::MetricRollupsService do
+  before do
+    allow(Settings.api).to receive(:metrics_default_limit).and_return(1000)
+  end
+
+  it 'validates that all of the parameters are present' do
```

Comment:
> I just felt that it was better to test that the validation is done here rather than add individual tests for the various subcollections. 

Definitely! I guess I was just thinking that the tests are telling me that validation is important enough to be its own service. If there's a responsibility besides validation that was important enough to extract, I think that behaviour deserves to also be tested at this level. It's OK to have some redundancy between tests of different sizes, since I'd expect a request spec to cover integration, and unit tests to cover all possible paths through the service